### DISCUSSION TITLE
Enquiry About the Upstream Health Check

### DISCUSSION THREAD

**Discussion by terryyrliang** [apache/apisix#6553] [Q&A] [answered]
Could you kindly advice if every apisix pod will send the health check request? Or will use some distributed lock to control only 1 pod will send the health check request? Thank you in advance.

**Comment by tzssangglass** [comment, marked as answer]
How do APISIX pods exchange the health status of upstream nodes with each other?

**Reply by terryyrliang** [reply]
Hi @tzssangglass . Thanks your answer. What I mean is if the one of the upstreams are not accessiable, and there are 2 apisix gateway pods, will these 2 pods trigger the health check schedule? Or they have some distributed lock to control only 1 pod will trigger the health check schedule.

**Reply by terryyrliang** [reply]
原来是南京的兄弟，我用中文说吧。 @tzssangglass  我的环境是这样的，有两个apisix gateway pod，然后一个routing配置了两个upstream。然后其中一个upstream挂掉了，那两个pod会同时去调度一个任务去检查那个挂掉的upstream吗？还是有一个分布式锁去控制这个调度只会发生在其中一个pod，然后会共享health check的结果呢？

**Reply by tzssangglass** [reply]
> will these 2 pods trigger the health check schedule

yes, two apisix pods are two APISIX instances that probe the upstream node independently.

> Or they have some distributed lock to control only 1 pod will trigger the health check schedule.

Currently there is no such feature.

**Reply by terryyrliang** [reply]
Big thanks to @tzssangglass . I will mark this as the answer. Thanks a lot.
